end2end-test-examples/gcs/src/main/java/io/grpc/gcs/JavaApiServicesClient.java

```diff
@@ -135,17 +137,15 @@ public void makeRandomMediaRequest(ResultTable results, int threadId) throws IOE
         req.setReturnRawInputStream(false);
         MediaHttpDownloader mediaHttpDownloader = req.getMediaHttpDownloader();
         mediaHttpDownloader.setDirectDownloadEnabled(true);
-        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
         req.executeMedia().download(byteArrayOutputStream);
-        buff.put(byteArrayOutputStream.toByteArray());
       } catch (IOException ioException) {
         System.err.println(ioException);
       }
       long dur = System.currentTimeMillis() - start;
-      if (buff.remaining() > 0) {
-        logger.warning("Got remaining bytes: " + buff.remaining());
+      if (byteArrayOutputStream.toByteArray().length > 0) {
+        logger.warning("Got remaining bytes: " + byteArrayOutputStream.toByteArray().length);
       }
-      buff.clear();
+      byteArrayOutputStream.close();
```

Comment:
rather than calling close, you probably want to move construction of the BAOS inside the for loop so that a fresh instance will be created for each run. `ByteArrayOutputStream#close()` is a no-op, and it will hold onto any already collected bytes (the next time around appending more to it).

As soon as the baos leaves scope (starting a new iteration of the for loop it will be free for garbage collection).